pkg/fileutils/fileutils.go

```diff
@@ -172,6 +172,9 @@ func (pm *PatternMatcher) MatchesOrParentMatches(file string) (bool, error) {
 // The "file" argument should be a slash-delimited path.
 //
 // MatchesUsingParentResult is not safe to call concurrently.
+//
+// Deprecated in favor of MatchesUsingParentResults: this function does behave
+// correctly in some cases (see https://github.com/docker/buildx/issues/850).
```

Comment:
This needs a colon after `Deprecated` to be picked up as "deprecated" https://github.com/golang/go/wiki/Deprecated

Perhaps something like (feel free to write something better 😂;

```suggestion
// Deprecated: this function does behave correctly in some cases (see https://github.com/docker/buildx/issues/850).
// Use MatchesUsingParentResults instead.
```

Reply:
Updated.

Note that actually deprecating the old functions has been challenging because of the circular dependencies. I wish we would fix these.